aim2dat/strct/strct_coordination.py

```diff
@@ -254,6 +267,43 @@ def _coord_calculate_n_nearest_neighbours(
     return _create_site_dict(structure, site_idx, neighbours)
 
 
+def _coord_calculate_atomic_radius(
+    site_idx,
+    structure,
+    r_max,
+    elements_sc,
+    positions_sc,
+    indices_sc,
+    mapping,
+    dist_matrix,
+    radius_type,
+    atomic_radius_delta,
+):
+    el_idx_sc = indices_sc.index(site_idx)
+    # min_dist = np.amin(np.delete(dist_matrix[site_idx], el_idx_sc))
+    position = np.array(structure.positions[site_idx])
+    site_radius = get_atomic_radius(structure.elements[site_idx], radius_type=radius_type)
+    if site_radius is None:
+        site_radius = get_atomic_radius(structure.elements[site_idx], radius_type="covalent")
```

Comment:
I think that this logic is not completely correct, or better to say, doesn't work as intended. You only check whether the radius is `None`, but the `ase.data` references return `nan` in case the value is not available. This can be seen in the following example for `Sc`.

```
from aim2dat.utils.element_properties import get_atomic_radius

radius = get_atomic_radius('Sc', 'vdw')
print(radius, radius is None) 
# nan False
```

As expected, the calculation of the coordination number for this scandium structure using the `vdw` radii would also fail:

```
from ase.build import bulk
from aim2dat.strct import Structure

sc = Structure.from_ase_atoms(bulk('Sc', 'hcp', a=3.3, c=5.2))
sc.calculate_coordination(method='atomic_radius', radius_type='vdw')

{'distance_avg': {},
 'distance_stdev': {},
 'distance_max': {},
 'distance_min': {},
 'nrs_avg': {('Sc', 'Sc'): 0},
 'nrs_stdev': {('Sc', 'Sc'): 0.0},
 'nrs_max': {('Sc', 'Sc'): 0},
 'nrs_min': {('Sc', 'Sc'): 0},
 'sites': [{'Sc': 0,
   'element': 'Sc',
   'kind': None,
   'position': [0.0, 0.0, 0.0],
   'neighbours': [],
   'total_cn': 0,
   'min_dist': 0.0,
   'max_dist': 0.0,
   'avg_dist': 0.0},
  {'Sc': 0,
   'element': 'Sc',
   'kind': None,
   'position': [-1.2952601953960163e-17, 1.9052558883257646, 2.6],
   'neighbours': [],
   'total_cn': 0,
   'min_dist': 0.0,
   'max_dist': 0.0,
   'avg_dist': 0.0}]}
```